src/Libraries/CoreNodesUI/Input/BasicInteractive.cs

```diff
@@ -22,8 +22,11 @@ public T Value
                     this.value = value;
                     if (!Equals(value, null))
                     {
-                        ForceReExecuteOfNode = true;
-                        OnAstUpdated();
+                        OnNodeModified();
+                    }
+                    else
+                    {
+                        ClearDirtyFlag();
                     }
```

Comment:
The above seems to be correct, just make sure we try out basic primitive type nodes with a watch node, and the value does get updated in watch node.